### PR TITLE
Add cargo keywords and categories.

### LINKED ISSUE #306: Add keywords to crate
I searched for a bignum crate on crates.io by typing "bignum" into the search bar. Several crates came up, but `num` was not one of them. I only found it by following a reference made to it in another crate.

Please can you add keywords to the crates.io entry, so `num` can be found when searching?

### PR DESCRIPTION
**Pull request by kw217**
Fixes #306. Suggestions welcome for better keywords/categories, though there is a limit of 5 of each.

### CODE REVIEW

**Comment by kw217** [comment]
This failure is surely unrelated to my change - the failures are
```
    src/float.rs - float::Float::is_sign_negative (line 348)
    src/float.rs - float::Float::is_sign_positive (line 329)
```

File: Cargo.toml [hunk 1]
```diff
@@ -3,7 +3,8 @@ authors = ["The Rust Project Developers"]
 description = "A collection of numeric types and traits for Rust, including bigint,\ncomplex, rational, range iterators, generic integers, and more!\n"
 documentation = "http://rust-num.github.io/num"
 homepage = "https://github.com/rust-num/num"
-keywords = ["mathematics", "numerics"]
+keywords = ["mathematics", "numerics", "bignum"]
+categories = [ "algorithms", "data-structures", "science", "cryptography" ]
```
Comment (cuviper) [inline Cargo.toml]: It seems a little bit of a stretch to say "science", but there aren't any math categories, so I guess that will do.

I'm less comfortable with "cryptography" though.  Yes, bigints are usually needed to implement crypto, but we don't implement any crypto algorithms here.

Comment (kw217) [inline Cargo.toml]: Thanks. Rebased, and I've added some suggestions to the subcrates. The [category slugs](https://crates.io/category_slugs) are a bit lacking, IMHO, and I also don't understand the purpose of all the subcrates, so again any suggestions are welcome!

**Comment by notriddle** [comment]
It's been rebased.

**Comment by cuviper** [comment]
Thanks!

bors r+

**Comment by kw217** [comment]
Thanks!

**Pull request merged**